app/web/features/auth/useAuthStore.ts

```diff
@@ -161,12 +161,50 @@ export default function useAuthStore() {
         }
         setLoading(false);
       },
+      async checkAuthStatus() {
+        setLoading(true);
+        try {
+          console.log("Checking auth status");
+          const res = await service.user.getAuthState();
+          if (res.loggedIn) {
+            console.log("Turns out we are logged in")!;
+            const auth = res.authRes!;
+            setUserId(auth.userId);
+            Sentry.setUser({ id: auth.userId.toString() });
+
+            //this must come after setting the userId, because calling setQueryData
+            //will also cause that query to be background fetched, and it needs
+            //userId to be set.
+            setJailed(auth.jailed);
+            setAuthenticated(true);
+          } else {
+            setAuthenticated(false);
+            setUserId(null);
+            Sentry.setUser({ id: undefined });
+          }
+        } catch (e) {
+          Sentry.captureException(e, {
+            tags: {
+              component: "auth/useAuthStore",
+              action: "checkAuthStatus",
+            },
+          });
+          setError(isGrpcError(e) ? e.message : fatalErrorMessage.current);
+        }
+        setLoading(false);
+      },
     }),
     //note: there should be no dependenices on the state or t, or
     //some useEffects will break. Eg. the token login in Login.tsx
     [setAuthenticated, setJailed, setUserId, setFlowState, queryClient]
   );
 
+  useEffect(() => {
+    if (!authenticated && typeof window !== "undefined") {
```

Comment:
when & how often could the variables `authActions` & `authenticated` change? this means, are we calling `checkAuthStatus` periodically?

Reply:
I don't actually know how that works. I thought this meant it rusn every time `useAuthStore` is loaded?

Reply:
like, this code runs every time the content of either `authenticated` or `authActions` change, then this code it's executed (and the first time this component is rendered).  

if this pretends to be executed periodically I don't know (by looking at the code) if this is achieved, I guess I have to debug/dig into it to check this

Reply:
if works it works, anyhow I can double check the behavior in my local set up